### PR TITLE
Fix: change type hints in specs to allow for floats in interval_days

### LINKED ISSUE #184: fix: change type hints in specs to allow for floats in interval_days
Currenty, float inputs to interval_days args in predictor and outcome specs are coerced into integers. Thus, it is not possible to generate predictors/outcomes with non-integer lookbehind/ahead windows. 

- [ ] Add test

### PR DESCRIPTION
**Pull request by bokajgd**
Fixes #184

### CODE REVIEW

File: src/timeseriesflattener/feature_spec_objects.py [hunk 1]
```diff
@@ -449,7 +450,7 @@ def get_col_str(self, additional_feature_name: Optional[str] = None) -> str:
         feature_name = self.feature_name
         if additional_feature_name:
             feature_name = feature_name + "-" + str(additional_feature_name)
-        col_str = f"{self.prefix}_{feature_name}_within_{self.interval_days}_days_{self.key_for_resolve_multiple}_fallback_{self.fallback}"
+        col_str = f"{self.prefix}_{feature_name}_within_{int(self.interval_days) if self.interval_days.is_integer() else str(self.interval_days).replace('.', '_')}_days_{self.key_for_resolve_multiple}_fallback_{self.fallback}"
```
Comment (MartinBernstorff) [inline src/timeseriesflattener/feature_spec_objects.py:453]: I think this is too much code to have within an `f-string`. Reading it is pretty hard. 

Try extracting `int(self.interval_days) if self.interval_days.is_integer() else str(self.interval_days).replace('.', '_')` into its own variable.

Also, ideally, we want each "unit" of information to be separated by `_`, as that's the current pattern.

Converting `1.3` days to `1_3_days` means that there's now an extra extra `_`, signifying an extra "unit" of information, which breaks some parsing I've written for e.g. `t2d-baseline-paper`. Is it possible to use another separator instead?

File: tests/test_timeseriesflattener/test_feature_spec_objects.py [hunk 2]
```diff
@@ -144,6 +144,18 @@ def test_resolve_multiple_fn_to_str():
     assert "maximum" in pred_spec_batch[0].get_col_str()
 
 
+def test_lookbehind_days_handles_floats():
+    """Test that resolve_multiple_fn is converted to str correctly."""
```
Comment (MartinBernstorff) [inline tests/test_timeseriesflattener/test_feature_spec_objects.py:148]: Docstring seems out of date? 😉

**Review by MartinBernstorff — APPROVED** [review]
Sorry about this mess Jakob, and thanks a tonne for the bugfix! I've added a few tiny comments, then it's ready for merge 👍

File: src/timeseriesflattener/feature_spec_objects.py (hunk 1 above)
Comment (bokajgd) [reply src/timeseriesflattener/feature_spec_objects.py]: Agree, with both points in this comment. Thanks for fixing the first point. Regarding another separator, I don't have any particular preferences. Perhaps use `-`?

File: tests/test_timeseriesflattener/test_feature_spec_objects.py (hunk 2 above)
Comment (bokajgd) [reply tests/test_timeseriesflattener/test_feature_spec_objects.py:149]: Oops. ;) Will you fix this as you're already committing minor changes?

Comment (MartinBernstorff) [reply tests/test_timeseriesflattener/test_feature_spec_objects.py:149]: Sure thing 👍

File: src/timeseriesflattener/feature_spec_objects.py (hunk 1 above)
Comment (MartinBernstorff) [reply src/timeseriesflattener/feature_spec_objects.py]: Sounds good 👍

**Pull request merged**